pkg/planner/core/rule_collect_plan_stats.go

```diff
@@ -274,11 +402,14 @@ func collectSyncIndices(ctx base.PlanContext,
 }
 
 func collectHistNeededItems(histNeededColumns []model.StatsLoadItem, histNeededIndices map[model.TableItemID]struct{}) (histNeededItems []model.StatsLoadItem) {
-	histNeededItems = make([]model.StatsLoadItem, 0, len(histNeededColumns)+len(histNeededIndices))
+	histNeededItems = histNeededColumns
+	if cap(histNeededItems) < len(histNeededColumns)+len(histNeededIndices) {
```

Comment:
I don't quite understand why we need to do this here. Can you explain it a little bit?